src/bridging/BridgingTypes.ts

```diff
@@ -0,0 +1,2831 @@
+  desktopAgent: string;
+}
+
+/**
+ * A desktop agent (typically for user channels) may want to provide additional information
+ * about how a channel can be represented in a UI. A common use case is for color linking.
+ */
+export interface DisplayMetadata {
+  color?: string;
+  glyph?: string;
+  name?: string;
+}
+
+/**
+ * Metadata relating to a single icon image at a remote URL, used to represent an
+ * application in a user interface.
+ */
+export interface Icon {
+  size?: string;
+  src: string;
+  type?: string;
+}
+
+/**
+ * Metadata relating to a single image at a remote URL, used to represent screenshot images.
+ */
+export interface Image {
+  label?: string;
+  size?: string;
+  src: string;
+  type?: string;
+}
+
+/**
+ * Metadata relating to the FDC3 DesktopAgent object and its provider
+ */
+export interface ImplementationMetadata {
+  appMetadata: AppMetadataElement;
+  fdc3Version: string;
+  optionalFeatures: ImplementationMetadataOptionalFeatures;
+  provider: string;
+  providerVersion?: string;
+}
+
+export interface ImplementationMetadataOptionalFeatures {
+  DesktopAgentBridging?: boolean;
+  OriginatingAppMetadata: boolean;
+  UserChannelMembershipAPIs: boolean;
+}
+
+/**
+ * The interface used to describe an intent within the platform.
+ */
+export interface IntentMetadata {
+  displayName: string;
+  name?: string;
+}
+
+/**
+ * IntentResolution provides a standard format for data returned upon resolving an intent.
+ */
+export interface IntentResolution {
+  intent: string;
+  source: SourceElement;
+  version?: string;
+}
+
+/**
+ * Describes results that an Intent handler may optionally return that should be
+ * communicated back to the app that raised the intent, via the IntentResolution.
+ *
+ * Represents a context channel that applications can join to share context data.
+ */
+export interface IntentResult {
+  id?: { [key: string]: any } | string;
+  name?: string;
+  type: string;
+  displayMetadata?: DisplayMetadataClass;
+  [property: string]: any;
+}
+
+export interface BridgeRequest {
+  meta: MetaObject;
+  /**
+   * The message payload typically contains the arguments to FDC3 API functions.
+   */
+  payload: { [key: string]: any };
+  /**
+   * Identifies the type of the message and it is typically set to the FDC3 function name that
+   * the message relates to, e.g. 'findIntent', with 'Request' appended.
+   */
+  type: string;
+}
+
+export interface MetaObject {
+  /**
+   * Optional field that represents the destination that the request should be routed to. Must
+   * be set by the Desktop Agent for API calls that include a target app parameter and must
+   * include the name of the Desktop Agent hosting the target application.
+   */
+  destination?: DestinationElement;
+  /**
+   * Unique GUID for the request
+   */
+  requestGuid: string;
+  /**
+   * Field that represents the source application that the request was received from.
+   */
+  source: DestinationElement;
```

Comment:
```suggestion
  source: SourceElement;
```
I think? Though I couldn't immediately see from the schema how this got generated

Reply:
We don't actually have control over this naming (which is often frustrating). Quicktype sometimes reuses elements its created elsewhere when they are identical, other times it doesn't (e.g. Why on earth do the context types generate both `Instrument` and `InstrumentElement`, which are identical), and then it does this elsewhere.

However, (and as per comments in the type generation  readme file) these names don't really matter. There needed to be a name for quicktype to reference when linking the parts of a top-level type, but you would rarely (I assume) ever refer to one of these directly, rather I think you'd type a message with the relevant top-level type. 

There are bugs in the recent versions of quicktype in the commandline argument parsing and we've not managed to
  find any combination of flags that improves on the output as yet. Hence if this ends up really needing to be solved we'll either need to go fix quicktype (which isn't the easiest piece of code to read...), to use different tooling or move to manually maintained sources. Perhsonally I'm least keen on manual maintenance... However, quite interested in your thought on this as you implement.

Reply:
Naming is now *much* improved through extensive work on the titles in every entity (we had more control over naming than first thought)